### PR TITLE
Fix missing progressive validation header

### PR DESCRIPTION
**Pull request by lalo**
At least when using the python vw api, the progressive validation header does not print. This will fix it.

### CODE REVIEW

File: vowpalwabbit/parse_args.cc [hunk 1]
```diff
@@ -1688,6 +1688,38 @@ void free_args(int argc, char* argv[])
   free(argv);
 }
 
+void print_progressive_validation_header(vw& all)
```
Comment (jackgerrits) [inline vowpalwabbit/parse_args.cc:1691]: Can this be const?

Comment (lalo) [reply vowpalwabbit/parse_args.cc:1691]: don't think so since we're using trace_message

File: cs/unittest/TestTracing.cs [hunk 2]
```diff
@@ -28,7 +28,8 @@ public void TestTraceListener()
             }
 
             var trace = string.Join("\n", messages);
-            Assert.AreEqual(16, messages.Count, $"Expected 16 lines. Found {messages.Count}. '{trace}'");
+            var expected = 19
```
Comment (lalo) [inline cs/unittest/TestTracing.cs:31]:
```suggestion
            var expected = 19;
```

File: cs/unittest/TestTracing.cs [hunk 3]
```diff
@@ -28,7 +28,8 @@ public void TestTraceListener()
             }
 
             var trace = string.Join("\n", messages);
-            Assert.AreEqual(16, messages.Count, $"Expected 16 lines. Found {messages.Count}. '{trace}'");
+            var expected = 19;
+            Assert.AreEqual(expected, messages.Count, $"Expected {expected} lines. Found {messages.Count}. '{trace}'");
             Assert.AreEqual("total feature number = 4", messages[15]);
```
Comment (lalo) [inline cs/unittest/TestTracing.cs:33]:
```suggestion
            Assert.AreEqual("total feature number = 4", messages[18]);
```

File: cs/unittest/TestTracing.cs [hunk 4]
```diff
@@ -28,8 +28,9 @@ public void TestTraceListener()
             }
 
             var trace = string.Join("\n", messages);
-            Assert.AreEqual(16, messages.Count, $"Expected 16 lines. Found {messages.Count}. '{trace}'");
-            Assert.AreEqual("total feature number = 4", messages[15]);
+            var expected = 19;
```
Comment (lalo) [inline cs/unittest/TestTracing.cs:31]: @lokitoth would this break something on the c# world?

Comment (lokitoth) [reply cs/unittest/TestTracing.cs:31]: I do not believe so. While there may be someone depending on the specific exact output, it seems unreasonable to expect there to never be new trace messages as the version increments.

It does not seem reasonable for consumers of this API to assume that the contents of the trace logging are an API contract.

**Review by jackgerrits — APPROVED** [review]

**Pull request merged**